README.rst

```diff
@@ -86,8 +86,7 @@ Using environment files (including any SSL settings) in ``~/.irods/``:
 ... except KeyError:
 ...     env_file = os.path.expanduser('~/.irods/irods_environment.json')
 ...
->>> ssl_context = ssl.create_default_context(purpose=ssl.Purpose.SERVER_AUTH, cafile=None, capath=None, cadata=None)
->>> ssl_settings = {'ssl_context': ssl_context}
+>>> ssl_settings = {'ssl_context': 'AUTO'}  # or provide an explicit SSLContext object
```

Comment:
As discussed this morning, we should consider removing the need for this.

If the user creates an SSL context, then they are responsible for configuring it correctly.

If the user defines SSL properties in their irods_environment.json file, the iRODSSession should read those values and construct an SSL context from them. This assumes the user does not pass an explicit SSL context. This also assumes they passed iRODSSession an environment file/object.

Reply:
>  This also assumes they passed iRODSSession an environment file/object.

That, or the equivalent configuration vars provided as keyword args in the iRODSSession constructor call.

Reply:
@trel  What @korydraughn said is right, I think.  We'll just not pass the ssl_context parameter when we wish the library to construct an SSLContext for us based on iRODS env vars. We could (should?) go all the way and have the new SSLContext code supplant the old. Since this is possibly a breaking change, maybe we should wait for a minor release, v1.2.0 - [per @alanking  suggestion.]

Reply:
But we could still enable users of the library to omit the ssl_context parameter entirely, to get the new behavior

Reply:
I'm confused as to what we're breaking? Did the old code ever do what users expected?

If the old implementation was broken, then that means we aren't breaking any existing behavior. Regardless, all of these proposals require the user to modify their code.

Please correct me if I'm misunderstanding something.

Reply:
@trel  @korydraughn. I could go either way.  I maintain there's an argument in support of either, and formally defer the question.

Reply:
Clarification: I was thinking "defer" until either release v1.2.0, or we've had a chance to talk about it with everyone present.  

I'm literally agnostic to which road we take because I think the old code is broken in ways we don't understand (but works in the ways our customers may presently find themselves relying on), but the new code is an improvement because it is based on a standard library function and is therefore - theoretically - either functional, or broken in ways we and the PSL developers  can understand).

Reply:
Ok, how's this? Subject to review. https://gist.github.com/d-w-moore/ed92b52da5f81751298bfee5d521ace8

Reply:
okay, after further discussion...

need to create/run test scenarios with different ways that the generated `ssl_context` can behave...

- if no breaking changes are found, we go with OBVIATE/REPLACE in a single release
- if a breaking scenario is found, we go with FALLBACK/AUTO_KEYWORD (and two releases, rather than one)

Reply:
Going with automatic generation of the object if the keyword argument isn't passed in.